### ISSUE TITLE
chatwoot SDK events

### ISSUE THREAD

**Issue by PyB1l** [chatwoot/chatwoot#3373]
Hi there, is there any event from the SDK that indicates that the agent has marked the consversasion as resloved?
I know there is a general `chatwoot:ready` that fires when `chatwootSDK.run` or after `$chatwoot.reset` methods, but i was
wondering if there are other events we can handle from SKD

**Comment by hiagodotme** [comment]
I need it too...

**Comment by sojan-official** [comment]
Could you share the specific events that you are looking for, as well as the use case? It will be super helpful for us to spec out this.

**Comment by PyB1l** [comment]
It would be great to have an event when the agent resolves the conversation so that that I could call `reset` method  and disable multiple user conversations from the same session

**Comment by hiagodotme** [comment]
Exactly, events like widget opened or widget closed is also interesting.

I'm already in production with chatwoot, I loved the software. I use it in a webview on my android app.

Knowing if the user clicked on the button to close the widget or else being able to disable it would be very good.

I am currently covering the close button with a div in absolute position.

**Comment by chris-garner** [comment]
It would also be great to have an event for things like `label_changed` or even better, build a mechanism for an agent to push any arbitrary event for a specific chat/context. For example, let's say an agent picked up a sales chat and they deem the lead to be of high value/likely to convert, it would be great if they could trigger an event (eg `likely_to_convert`) that allows the client-side to subsequently push a conversion event out to Google Ads/GTM etc. Thus helping to optimise and refund ads data and bidding strategies etc.

**Comment by sojan-official** [comment]
ref: https://github.com/chatwoot/chatwoot/issues/3373

**Comment by sojan-official** [comment]
> This would be a great feature to have... or at least a way to manually check agent availability via the JS SDK, we'd instead either hide the chat icon or display a "submit a ticket" icon that links to our clientarea.

ref: https://github.com/chatwoot/chatwoot/issues/273#issuecomment-917901179

**Comment by JWesorick** [comment]
Would be nice to have events for when they begin a conversation or send a message to track in front end analytics.

**Comment by julienmarie** [comment]
I more than second this. Being able to track start of conversation and messages, identify events (when a customer enters his email address by example), and also when a conversation is resolved is key for us in terms of analytics so we can correlate impact on conversion rate on our e-commerce properties. It's quite a limiting factor and we would need to resort to complicated workarounds without it.

**Comment by stonecharioteer** [comment]
This issue is being closed due to lack of activity for over a year. [For more details, visit this discussion.](https://github.com/orgs/chatwoot/discussions/12810)
